### DISCUSSION TITLE
Property binding for structs

### DISCUSSION THREAD

**Discussion by isaac-eiter** [slint-ui/slint#2520] [General] [answered]
I have a struct with multiple nested structs, simplified example:
```
struct Region {
    top_left: Point,
    top_right: Point,
    bottom_left: Point,
    bottom_right: Point,
}

struct ProjectSettings {
    name: string,
    bounding_box: Region, 
}
```

I want to be able to do a two-way binding from my `MainWindow` to a Settings specific page, and pass those values downwards in a way that "fans out" so that lower level objects in the tree don't need to expose all of their sub-properties for definition in the top-level Window definition, but can just expose the property as a struct. This example code does not result in actual application logic settings changing:

```
export component MainWindow inherits Window {
    title: "Demo";

    in-out property<ProjectSettings> settings <=> settings-page.settings;

    settings-page := SettingsPage{}

}

export component SettingsPage {
    in-out property <ProjectSettings> settings;

    LineEdit {
        text: settings.bounding_box.top_left;
    }
    // add rest...
}
```

My problem with this code is using the `LineEdit` doesn't result in the settings values actually changing, which I expect is because the values aren't bound together so the text input changes but nothing gets rebuilt/passed on. Is there a way to do this? Ideally it would be nice to be able to define bindings to multiple objects, maybe something like:

```
export component MainWindow inherits Window {
    title: "Demo";

    in-out property<ProjectSettings> settings <=> settings-page.settings;

    settings-page := SettingsPage{}

}

export component SettingsPage {
    in-out property <ProjectSettings> settings <=>  {name.text, bounding-box.box};

   name := LineEdit {}
   bounding-box := BoundingBoxEntry{}
}

export component BoundingBoxEntry {
    in-out property box <=> {top_left:  entry-top-left.text, top_right: entry-top-right.text, ... }
    entry-top-left := LineEdit {}
    entry-top-right := LineEdit {}
    /* ... add bottom left/right ... */
}
```

**Comment by ogoffart** [comment, marked as answer]
Unfortunatelt, we don't support binding to memeber of the struct yet.
We might support that int the future,  there is an issue to track it: https://github.com/slint-ui/slint/issues/814

In the mean time, wou will have to do some manual work:  Use the `accepted` or `edited` callback of the LineEdit to notify your backend.
And add some functions in your ui to reload the config.

```

export component MainWindow inherits Window {
    title: "Demo";

    in-out property<ProjectSettings> settings <=> settings-page.settings;

    settings-page := SettingsPage{}

    public function reload_settings() { settings-page.reload_settings(); }
}

export component SettingsPage {
    in-out property <ProjectSettings> settings;

    le := LineEdit {
        text: settings.bounding_box.top_left;
        accepted => { settings.bounding_box.top_left = text; }
    }

    public function reload_settings() { le.text =  settings.bounding_box.top_left;}
}
```

(something like that)
